chef/automate
# Update IAMv2 project tags in nodemanager

### :nut_and_bolt: Description: What code changed, and why?
Adding the node manager to the IAMv2 project update. 

### :chains: Related Resources
#536

### :+1: Definition of Done
Project tags for nodes in the nodemanager are updated during a project update event. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-deployment && build components/authz-service && build components/nodemanager-service`
1. Upgrade to beta2.1 and load data `chef-automate iam upgrade-to-v2 --beta2.1`
1. Add some compliance nodes with `chef_load_compliance_nodes 5`
1. Add some client runs nodes with `chef_load_nodes 5`
1. Create a project with rules
1. Check PostgreSQL that the projects on the nodes were updated. To look at postgres use `chef-automate dev psql nodemanager_service`. The query I have been using is `select n.id, COALESCE(array_to_json(array_remove(array_agg(p.project_id), NULL)), '[]') AS projects, n.projects_data from nodes n LEFT JOIN nodes_projects np on np.node_id = n.id LEFT JOIN projects p on np.project_id = p.id group by n.id;`

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
